pages/index.js

```diff
@@ -1,5 +1,32 @@
 import PlantList from "@/components/PlantList";
+import { useState } from "react";
+import styled from "styled-components";
 
-export default function Homepage() {
-  return <PlantList />;
+export default function Homepage({ plants, toggleFavourite, onSubmitPlant }) {
+  const [showForm, setShowForm] = useState(false);
+  return (
+    <>
+      <StyledHeader>My Plants Collection</StyledHeader>
+      <ButtonContainer>
+        <button onClick={() => setShowForm((prev) => !prev)}>
+          {showForm ? "Hide form" : "Add new plant"}
+        </button>
+      </ButtonContainer>
```

Comment:
Here where you have the button, you should render the Form conditionally:
```js
{ showForm && <PlantForm .../> }
```

Best practice is also to define a handler function 'handleCloseForm()' that can be passed to the Form component.